core/tests/wdio.conf.js

```diff
@@ -0,0 +1,414 @@
+    // resolved to continue.
+    /**
+     * Gets executed once before all workers get launched.
+     * @param {Object} config wdio configuration object
+     * @param {Array.<Object>} capabilities list of capabilities details
+     */
+    //  onPrepare: function(config, capabilities) {
+      
+    // },
+    /**
+     * Gets executed before a worker process is spawned and can be used to initialise specific service
+     * for that worker as well as modify runtime environments in an async fashion.
+     * @param  {String} cid      capability id (e.g 0-0)
+     * @param  {[type]} caps     object containing capabilities for session that will be spawn in the worker
+     * @param  {[type]} specs    specs to be run in the worker process
+     * @param  {[type]} args     object that will be merged with the main configuration once worker is initialized
+     * @param  {[type]} execArgv list of string arguments passed to the worker process
+     */
+    // onWorkerStart: function (cid, caps, specs, args, execArgv) {
+    // },
+    /**
+     * Gets executed just after a worker process has exited.
+     * @param  {String} cid      capability id (e.g 0-0)
+     * @param  {Number} exitCode 0 - success, 1 - fail
+     * @param  {[type]} specs    specs to be run in the worker process
+     * @param  {Number} retries  number of retries used
+     */
+    // onWorkerEnd: function (cid, exitCode, specs, retries) {
+    // },
+    /**
+     * Gets executed just before initialising the webdriver session and test framework. It allows you
+     * to manipulate configurations depending on the capability or spec.
+     * @param {Object} config wdio configuration object
+     * @param {Array.<Object>} capabilities list of capabilities details
+     * @param {Array.<String>} specs List of spec file paths that are to be run
+     * @param {String} cid worker id (e.g. 0-0)
+     */
+    // beforeSession: function (config, capabilities, specs, cid) {
+    // },
+    /**
+     * Gets executed before test execution begins. At this point you can access to all global
+     * variables like `browser`. It is the perfect place to define custom commands.
+     * @param {Array.<Object>} capabilities list of capabilities details
+     * @param {Array.<String>} specs        List of spec file paths that are to be run
+     * @param {Object}         browser      instance of created browser/device session
+     */
+    before: function() {
+      // Set a wide enough window size for the navbar in the library pages to
+      // display fully.
+      browser.setWindowSize(1285, 1000);
+
+
+      // Configure the Firebase Admin SDK to communicate with the emulator.
+      process.env.FIREBASE_AUTH_EMULATOR_HOST = 'localhost:9099';
+      FirebaseAdmin.initializeApp({projectId: 'dev-project-id'});
+    
+      // Navigate to the splash page so that tests can begin on an Angular page.
+      browser.url('http://localhost:9001');
```

Comment:
Ditto for this. This should be most probably in onPrepare hook.

Reply:
As already mentioned browser is not defined when this hook is executed so we cannot add it here.

Reply:
Can we not move this to the before hook?

> Gets executed before test execution begins. At this point you can access to all global
    variables like `browser`. It is the perfect place to define custom commands.

Reply:
its already inside the before hook.